tests/uri.cpp

```diff
@@ -288,3 +288,13 @@ BOOST_AUTO_TEST_CASE(test_host_port_without_schema)
     BOOST_CHECK_EQUAL( uri3.getHost(), "host" );
     BOOST_CHECK_EQUAL( uri3.getPort(), 12345 );
 }
+
+BOOST_AUTO_TEST_CASE(query_without_value)
+{
+    const servus::URI uri( "?foo,bar=foo,blubb" );
```

Comment:
=foo will be ignored, which is reasonable. Another option will be throwing.
foo= should behave the same as foo, which is a key with an empty string as value. The code seems to do that but I wonder if std::string("foo").substr(3) works or crashes.